src/main/java/io/singularitynet/MissionHandlers/ObservationFromItemsImplementation.java

```diff
@@ -0,0 +1,74 @@
+package io.singularitynet.MissionHandlers;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import io.singularitynet.MissionHandlerInterfaces.ICommandHandler;
+import io.singularitynet.MissionHandlerInterfaces.IObservationProducer;
+import io.singularitynet.projectmalmo.MissionInit;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.item.Item;
+import net.minecraft.util.registry.Registry;
+
+import io.singularitynet.projectmalmo.ObservationFromItem;
+import io.singularitynet.projectmalmo.ObservationFromItems;
+
+import java.util.List;
+
+import static net.minecraft.util.registry.Registry.ITEM_KEY;
+
+public class ObservationFromItemsImplementation extends HandlerBase implements IObservationProducer, ICommandHandler {
+    private boolean sendRec;
+
+    @Override
+    public boolean isOverriding() {
+        return false;
+    }
+
+    @Override
+    public void setOverriding(boolean b) {}
+
+    @Override
+    public void install(MissionInit currentMissionInit) { sendRec = false; }
+
+    @Override
+    public void deinstall(MissionInit currentMissionInit) {}
+
+    @Override
+    public boolean execute(String command, MissionInit currentMissionInit) {
+        String comm[] = command.split(" ", 2);
+        if (comm.length == 2 && comm[0].equalsIgnoreCase(ObservationFromItem.ITEM_LIST.value()) &&
+                !comm[1].equalsIgnoreCase("off")) {
+            this.sendRec = true;
+            return true;
+        }
+        if (comm.length == 2 && comm[0].equalsIgnoreCase(ObservationFromItem.ITEM_LIST.value()) &&
+                comm[1].equalsIgnoreCase("off")
+        ) {
+            this.sendRec = false;
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void cleanup() {}
+
+    @Override
+    public void prepare(MissionInit missionInit) {}
+
+    @Override
+    public void writeObservationsToJSON(JsonObject json, MissionInit currentMissionInit) {
+        if (!this.sendRec){
+            return;
+        }
+        Registry<Item> str_ent = MinecraftClient.getInstance().world.getRegistryManager().get(ITEM_KEY);
+        List<Item> list_ent = str_ent.stream().toList();
+        JsonArray items = new JsonArray();
+        for (Item ent: list_ent)
+        {
+            String item_name = ent.toString();
+            items.add(item_name);
+        }
+        json.add("item_list", items);
```

Comment:
Does it contain blocks or just items that might be crafted?

Reply:
@noskill  actually, I'm not sure. List does contain some block-like things like flower_pot, air, oak_sign, but not [Acacia Wall Sign](https://minecraft.fandom.com/wiki/Acacia_Wall_Sign) for example. But it contains craftable items like pickaxes, iron_ingot.